build-logic/build.gradle.kts

```
@@ -50,9 +51,15 @@ dependencies {
   runtimeOnly(golatac.lib("kotlinx.binarycompatibilityvalidator"))
 }
 
+// Keep in sync with CompilerOptions.kt
 java {
-  // Keep in sync with CompilerOptions.kt
-  toolchain.languageVersion.set(JavaLanguageVersion.of(11))
+  toolchain.languageVersion.set(JavaLanguageVersion.of(17))
+}
+tasks.withType(JavaCompile::class.java).configureEach {
+  options.release.set(11)
+}
```

FYI there is [Kotlin DSL syntax](https://gradle.github.io/kotlin-dsl-docs/api/org.gradle.kotlin.dsl/org.gradle.api.-domain-object-collection/index.html) with nice Kotlin generics. I'm not using .kts but I think it's available there
```suggestion
tasks.withType<JavaCompile>().configureEach {
  options.release.set(11)
}
```